pkg/asset/agent/image/baseiso.go

```diff
@@ -40,7 +37,7 @@ func (i *BaseIso) Name() string {
 }
 
 // getIsoFile is a pluggable function that gets the base ISO file
-type getIsoFile func() (string, error)
+type getIsoFile func(archName string) (string, error)
```

Comment:
It's tempting to make the argument here a `types.Architecture` and push the conversion down into `downloadIso()`. That way we get the compiler checking the types for us and in code reviews we only have to be on the lookout for dodgy casts.

Reply:
Ah, true, we'd have to cast it somewhere to get this to work, which kind of eliminates the benefit.